nnsvs/model.py

```diff
@@ -72,14 +71,47 @@ def __init__(self, in_dim, hidden_dim, out_dim, num_layers=1, bidirectional=True
         super(LSTMRNN, self).__init__()
         self.hidden_dim = hidden_dim
         self.num_layers = num_layers
+        self.relu = nn.ReLU()
         self.num_direction =  2 if bidirectional else 1
         self.lstm = nn.LSTM(in_dim, hidden_dim, num_layers,
             bidirectional=bidirectional, batch_first=True, dropout=dropout)
         self.hidden2out = nn.Linear(self.num_direction*self.hidden_dim, out_dim)
-
+        self.prediction_type="deterministic"
     def forward(self, sequence, lengths):
         sequence = pack_padded_sequence(sequence, lengths, batch_first=True)
         out, _ = self.lstm(sequence)
         out, _ = pad_packed_sequence(out, batch_first=True)
         out = self.hidden2out(out)
         return out
+
+class RMDN(nn.Module):
+    def __init__(self, in_dim, hidden_dim, out_dim, num_layers=1, bidirectional=True, dropout=0.0, num_gaussians=8):
+        super(RMDN, self).__init__()
+        self.linear = nn.Linear(in_dim, hidden_dim)
+        self.relu = nn.ReLU()
+        self.num_direction=2 if bidirectional else 1
+        self.lstm = nn.LSTM(hidden_dim, hidden_dim, num_layers, 
+                            bidirectional=bidirectional, batch_first=True, 
+                            dropout=dropout)
+        self.mdn = MDNLayer(self.num_direction*hidden_dim, out_dim, num_gaussians=num_gaussians)
+        self.prediction_type="probabilistic"
```

Comment:
Same as above.

Reply:
I changed to use  predictin_type() of BaseModel in the latest commit.